### PR TITLE
Rename To ICM

### PR DESCRIPTION
**Pull request by geoff-vball**
## Why this should be merged
Closes https://github.com/ava-labs/teleporter/issues/620

It's probably easier to view this PR commit by commit.

There are a few quirks of the renaming. Now that `Warp` and `Teleporter` are both `ICM`, it's hard to document `Teleporter`. In most spots, I opted for specifying `TeleporterMessenger` the contract, to disambiguate it from `ICM at large`. 

There are some places where our upstream dependencies haven't renamed `subnet` fields to `l1`, are some of them going to stay that way? Is warp going to get renamed in the dependencies?

### CODE REVIEW

File: contracts/ictt/README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ A concrete implementation of `TokenHome` and `INativeTokenTransferrer` that hand
...
+An abstract implementation of `ITokenTransferrer` for a token transfer contract on a "remote" chain that receives transferred assets from a specific `TokenHome` instance. Each `TokenRemote` instance has a single `TokenHome` instance that it receives token transfers from to mint tokens. It also handles sending messages (and correspondingly burning tokens) to route tokens back to other chains (either its `TokenHome`, or other `TokenRemote` instances). Once deployed, a `TokenRemote` instance must be registered with its specified `TokenHome` contract. This is done by calling `registerWithHome` on the remote contract, which will send a ICM message to the home contract with the information to register.
```
Comment (richardpringle) [inline contracts/ictt/README.md:55]:
```suggestion
An abstract implementation of `ITokenTransferrer` for a token transfer contract on a "remote" chain that receives transferred assets from a specific `TokenHome` instance. Each `TokenRemote` instance has a single `TokenHome` instance that it receives token transfers from to mint tokens. It also handles sending messages (and correspondingly burning tokens) to route tokens back to other chains (either its `TokenHome`, or other `TokenRemote` instances). Once deployed, a `TokenRemote` instance must be registered with its specified `TokenHome` contract. This is done by calling `registerWithHome` on the remote contract, which will send an ICM message to the home contract with the information to register.
```

Comment (richardpringle) [reply contracts/ictt/README.md:55]: (if this is the only change request, can do it in a follow up haha)

File: contracts/ictt/README.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ To account for the need to bootstrap the chain using a transferred asset as its
...
 2. Deploy the `NativeTokenRemote` contract to the pre-derived address set in the blockchain configuration of step 1. The `initialReserveImbalance` should be 100, matching the number of tokens allocated in the genesis block that were not initially backed by collateral in the `TokenHome` instance.
-3. Call the `registerWithHome` function on the `NativeTokenRemote` instance to send a Teleporter message registering this remote with its `TokenHome`. This message should be relayed and delivered to the `TokenHome` instance.
+3. Call the `registerWithHome` function on the `NativeTokenRemote` instance to send a ICM message registering this remote with its `TokenHome`. This message should be relayed and delivered to the `TokenHome` instance.
```
Comment (richardpringle) [inline contracts/ictt/README.md:77]:
```suggestion
3. Call the `registerWithHome` function on the `NativeTokenRemote` instance to send an ICM message registering this remote with its `TokenHome`. This message should be relayed and delivered to the `TokenHome` instance.
```

File: contracts/validator-manager/ValidatorManager.sol [hunk 3]
```diff
@@ -118,7 +118,7 @@ abstract contract ValidatorManager is Initializable, ContextUpgradeable, IValida
         onlyInitializing
     {
         ValidatorManagerStorage storage $ = _getValidatorManagerStorage();
-        $._subnetID = settings.subnetID;
+        $._l1ID = settings.l1ID;
```
Comment (richardpringle) [inline contracts/validator-manager/ValidatorManager.sol:121]: 😂 this hurts my eyes

File: contracts/validator-manager/ValidatorManager.sol [hunk 4]
```diff
@@ -168,8 +168,8 @@ abstract contract ValidatorManager is Initializable, ContextUpgradeable, IValida
             }
 
             // Validation ID of the initial validators is the sha256 hash of the
-            // convert Subnet tx ID and the index of the initial validator.
-            bytes32 validationID = sha256(abi.encodePacked(conversionData.subnetID, i));
+            // convert L1 to L1 tx ID and the index of the initial validator.
```
Comment (richardpringle) [inline contracts/validator-manager/ValidatorManager.sol:171]: "convert L1 to L1"?

**Review by richardpringle — APPROVED** [review]
Honestly... the more time I stare at all these changes, the less likely I am to catch something that's off. I think we should just merge this TBH. If there are mistakes or missed changes (which wouldn't show up in this PR anyway), we can change them as needed.

**Comment by richardpringle** [comment]
🤔  all these files still have the word `warp` in them... is that okay?
```
tests/network/network.go
tests/utils/warp-genesis-template.json
tests/utils/governance.go
tests/utils/chain.go
tests/utils/validator_manager.go
tests/utils/teleporter.go
tests/flows/teleporter/validator_churn.go
tests/flows/teleporter/registry/teleporter_registry.go
tests/flows/teleporter/relayer_modifies_message.go
tests/flows/validator-manager/erc20_token_staking.go
tests/flows/validator-manager/native_token_staking.go
tests/suites/governance/governance_suite_test.go
tests/suites/ictt/ictt_suite_test.go
tests/suites/teleporter/teleporter_suite_test.go
tests/suites/validator-manager/validator_manager_suite_test.go
utils/gas-utils/gas_utils.go
contracts/teleporter/registry/UPGRADING.md
contracts/teleporter/registry/README.md
contracts/teleporter/TeleporterMessenger.sol
contracts/teleporter/ITeleporterMessenger.sol
contracts/teleporter/registry/TeleporterRegistry.sol
contracts/teleporter/tests/RedeemRelayerRewardsTests.t.sol
contracts/teleporter/tests/SendSpecifiedReceiptsTests.t.sol
contracts/teleporter/tests/MarkReceiptTests.t.sol
contracts/teleporter/tests/InitializeBlockchainIDTests.t.sol
contracts/teleporter/tests/GetOutstandingReceiptsToSendTests.t.sol
contracts/teleporter/tests/GetNextMessageIdTests.t.sol
contracts/teleporter/registry/tests/UpdateMinTeleporterVersionTests.t.sol
contracts/teleporter/tests/RetryMessageExecutionTests.t.sol
contracts/teleporter/tests/FallbackReceiveTests.t.sol
contracts/teleporter/tests/HandleInitialMessageExecutionTests.t.sol
contracts/teleporter/tests/TeleporterMessengerTest.t.sol
contracts/teleporter/registry/tests/NonReentrantTests.t.sol
contracts/teleporter/tests/SendCrossChainMessageTests.t.sol
contracts/ictt/TokenHome/TokenHome.sol
contracts/teleporter/tests/ReceiveCrossChainMessageTests.t.sol
contracts/teleporter/registry/tests/TeleporterRegistryTests.t.sol
abi-bindings/go/validator-manager/NativeTokenStakingManager/NativeTokenStakingManager.go
abi-bindings/go/governance/ValidatorSetSig/ValidatorSetSig.go
abi-bindings/go/validator-manager/ERC20TokenStakingManager/ERC20TokenStakingManager.go
abi-bindings/go/validator-manager/PoAValidatorManager/PoAValidatorManager.go
contracts/governance/README.md
contracts/ictt/tests/TokenRemoteTests.t.sol
contracts/ictt/tests/TokenHomeTests.t.sol
contracts/ictt/TokenRemote/TokenRemote.sol
abi-bindings/go/teleporter/TeleporterMessenger/TeleporterMessenger.go
abi-bindings/go/teleporter/registry/TeleporterRegistry/packing.go
contracts/ictt/tests/TokenTransferrerTests.t.sol
abi-bindings/go/teleporter/registry/TeleporterRegistry/TeleporterRegistry.go
contracts/governance/tests/ValidatorSetSigTests.t.sol
contracts/governance/ValidatorSetSig.sol
abi-bindings/go/teleporter/registry/TeleporterRegistry/packing_test.go
contracts/validator-manager/README.md
contracts/validator-manager/PoSValidatorManager.sol
contracts/validator-manager/UptimeMessageSpec.md
contracts/validator-manager/ValidatorManager.sol
cmd/teleporter-cli/transaction.go
contracts/validator-manager/tests/PoSValidatorManagerTests.t.sol
contracts/validator-manager/tests/ValidatorManagerTests.t.sol
```

File: contracts/validator-manager/ValidatorManager.sol (hunk 3 above)
Comment (geoff-vball) [reply contracts/validator-manager/ValidatorManager.sol:121]: I don't disagree...

**Comment by iansuvak** [comment]
Are we renaming the module in `go.mod`?

**Review by iansuvak — APPROVED** [review]

**Pull request merged**